caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java

```diff
@@ -0,0 +1,273 @@
+/*
+ * Copyright 2015-2017 Hewlett Packard Enterprise Development LP.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hpe.caf.auditing.webserviceclient;
+
+import com.hpe.caf.auditing.AuditCoreMetadataProvider;
+import com.hpe.caf.auditing.AuditEventBuilder;
+import com.hpe.caf.auditing.elastic.ElasticAuditConstants;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentFactory;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.List;
+
+public class WebserviceClientAuditEventBuilder implements AuditEventBuilder {
+
+    private static final Logger LOG = LogManager.getLogger(WebserviceClientAuditEventBuilder.class.getName());
+
+    private final String webserviceHostAndPort;
+
+    private final Map<String, Object> auditEventCommonFields = new HashMap<>();
+
+    private final List<EventParam> auditEventParams = new ArrayList<>();
+
+    private XContentBuilder jsonBuilder;
+
+    public WebserviceClientAuditEventBuilder(final String webserviceHostAndPort,
+                                             final AuditCoreMetadataProvider coreMetadataProvider) {
+        this.webserviceHostAndPort = webserviceHostAndPort;
+
+        try {
```

Comment:
Corrected in the latest commit. I missed it as I went over cleanup in the code; went I first started using the JSON builder I was using it differently that how it is being used in the code now and I required it in the constructor.